.github/workflows/ci.yml

```diff
@@ -27,52 +27,38 @@ concurrency:
 jobs:
   test:
     if: "!contains(github.event.head_commit.message, 'skip ci')"
-    name: ${{ matrix.os }} - ${{ matrix.p4est_test }} - Julia ${{ matrix.version }}
+    name: ${{ matrix.os }} - ${{ matrix.JULIA_P4EST_TEST }} - Julia ${{ matrix.version }}
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
         version:
           - '1.6'
+          - '1'
           # - 'nightly'
         os:
           - ubuntu-latest
-          # - macos-latest
-          # - windows-latest
+          - macos-latest
+          - windows-latest
         arch:
           - x64
-        p4est_test:
-          - P4EST_JLL_USES_MPI_PRE_GENERATED_BINDINGS
-          - P4EST_JLL_USES_MPI
-          - P4EST_CUSTOM_NON_MPI
-          - P4EST_CUSTOM_USES_MPI
-        # Only selected tests on macOS and Windows, since not everything works everywhere yet
-        include:
+        JULIA_P4EST_TEST:
+          - P4EST_JLL_MPI_DEFAULT
+          - P4EST_CUSTOM_MPI_CUSTOM
+        exclude:
+          # Some preferences are not loaded in package tests in older versions
+          # of Julia, see
+          # https://github.com/JuliaPackaging/Preferences.jl/issues/40
+          # Thus, we need to disable this setup in CI.
           - version: '1.6'
-            os: macOS-latest
-            arch: x64
-            p4est_test: P4EST_JLL_USES_MPI_PRE_GENERATED_BINDINGS
-          - version: '1.6'
-            os: macOS-latest
-            arch: x64
-            p4est_test: P4EST_JLL_USES_MPI
-          - version: '1.6'
-            os: windows-latest
-            arch: x64
-            p4est_test: P4EST_JLL_USES_MPI_PRE_GENERATED_BINDINGS
-          # Additional tests on Julia v1.7
-          - version: '^1.7.0-0'
-            os: ubuntu-latest
-            arch: x64
-            p4est_test: P4EST_JLL_USES_MPI_PRE_GENERATED_BINDINGS
-          - version: '^1.7.0-0'
-            os: macOS-latest
-            arch: x64
-            p4est_test: P4EST_JLL_USES_MPI_PRE_GENERATED_BINDINGS
-          - version: '^1.7.0-0'
-            os: windows-latest
-            arch: x64
-            p4est_test: P4EST_JLL_USES_MPI_PRE_GENERATED_BINDINGS
+            JULIA_P4EST_TEST: P4EST_CUSTOM_MPI_CUSTOM
+          # Run only selected tests on macOS and Windows, since not everything
+          # works everywhere yet and we are most familiar with setting up custom
+          # p4est and MPI builds on Linux.
+          - os: macos-latest
+            JULIA_P4EST_TEST: P4EST_CUSTOM_MPI_CUSTOM
+          - os: windows-latest
+            JULIA_P4EST_TEST: P4EST_CUSTOM_MPI_CUSTOM
     steps:
       - uses: actions/checkout@v3
       - name: Set p4est release to run tests against
```

Comment:
Not shown here but a few lines below: Shouldn't we test against `P4EST_RELEASE=2.8` and not `2.3.1`?

Reply:
Please see the comment I added in d917d77

Reply:
🙏